static/app/utils/profiling/flamegraph/flamegraphStateProvider/index.tsx

```diff
@@ -280,3 +329,36 @@ export function FlamegraphStateProvider(
     </FlamegraphStateContext.Provider>
   );
 }
+
+type Candidate = {
+  score: number;
+  threadId: number | null;
+};
+
+function scoreFlamegraph(
+  flamegraph: Flamegraph,
+  focusFrame: FlamegraphProfiles['focusFrame']
+): number {
+  if (!defined(focusFrame)) {
+    return 0;
+  }
+
+  let score = 0;
+
+  const frames: FlamegraphFrame[] = [...flamegraph.root.children];
+  while (frames.length > 0) {
+    const frame = frames.pop()!;
+    if (
+      frame.frame.name === focusFrame.name &&
+      frame.frame.image === focusFrame.package
+    ) {
+      score += frame.node.totalWeight;
```

Comment:
It looks like we are trying to find a max over different frames right? In that case, do we want to favor longer running frames vs multiple very fast ones?

Reply:
this was just a simple scoring function to find the thread that spent the most time in a particular function, i think the more complex scoring should come later if they're necessary